core/src/main/scala/io/finch/Input.scala

```diff
@@ -10,6 +13,31 @@ final case class Input(request: Request, path: Seq[String]) {
   def headOption: Option[String] = path.headOption
   def drop(n: Int): Input = copy(path = path.drop(n))
   def isEmpty: Boolean = path.isEmpty
+
+  def withBody(buf: Buf, charset: Option[Charset] = None): Input = {
+    request.content = buf
+    request.contentLength = buf.length.toLong
+    if (charset.isDefined) request.charset = charset.get.displayName()
+    this
+  }
+
+  def withHeaders(headers: (String, String)*): Input = {
+    headers.foreach { case (k, v) => request.headerMap.set(k, v) }
+    this
+  }
+
+  def withForm(params: (String, String)*): Input = {
+    if (params.nonEmpty) {
+      val req = RequestBuilder()
+        .addFormElement(params: _*)
+        .addHeaders(request.headerMap.toMap)
+        .url(request.host.get)
+        .buildFormPost()
+      copy(request = req)
+    } else {
+      this
+    }
```

Comment:
I'm seeing a weird NPE from finagle's when `params` is an empty seq (cf [buildFormPost](https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/RequestBuilder.scala#L347-L383).

I guess netty's `HttpPostRequestEncoder` does not like having no body.